### PR TITLE
fix wrong results came from a bug in skip optimization

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: src/inverted_index.c [hunk 1]
```diff
@@ -561,16 +561,11 @@ SKIPPER(seekFreqOffsetsFlags) {
       qint_decode4(br, &did, &freq, (uint32_t *)&fm, &offsz);
       Buffer_Skip(br, offsz);
       lastId = (did += lastId);
-      if (did > expid) {
+      if (!(ctx->num & fm)) {
```
Comment (mnunberg) [inline src/inverted_index.c]: Don't dereference ctx; first check did >= expid as before

Comment (MeirShpilraien) [reply src/inverted_index.c]: This was actually the issue, you did not check that the doc not match the field mask

Comment (mnunberg) [reply src/inverted_index.c]: hrm, I see. Can you cache the variable `ctx->num` as `userFm` locally, and use that? I'd also like to see some comments regarding why skipper is called in a loop (it took me a while to figure out why...)

**Comment by Nomafin** [comment]
Hey, any chance you can make some sort of pre-release-version with this fix? We are suffering from this bug in a production version of our app and even a slightly less performant version with right search results would be great.

**Comment by mnunberg** [comment]
@Nomafin You're free to compile it and load it; did yo mean a docker image or similar?

File: src/inverted_index.c (hunk 1 above)
Comment (MeirShpilraien) [reply src/inverted_index.c]: I think the compiler probably optimize it anyway, but yeah I will change it...

Comment (MeirShpilraien) [reply src/inverted_index.c]: By the way a much better optimization would be to use a binary search on the blocks.

**Comment by MeirShpilraien** [comment]
@mnunberg I fixed the review.
@Nomafin We will release 1.6.4 with this fix soon..

File: src/inverted_index.c (hunk 1 above)
Comment (mnunberg) [reply src/inverted_index.c]: We already do a binary search on the block to find which block contains the ID, but within the individual block we must read sequentially.

Comment (MeirShpilraien) [reply src/inverted_index.c]: From looking at the code, not binary search is done ..

Comment (MeirShpilraien) [reply src/inverted_index.c]: Sorry, found it :+1:

**Comment by Nomafin** [comment]
> @Nomafin You're free to compile it and load it; did yo mean a docker image or similar?

Yeah, we use the Docker image.

**Comment by mnunberg** [comment]
@Nomafin You should be able to use `redisearch:edge` which corresponds to the latest master version, as this change is now merged

**Pull request merged**